src/usecases/counter/providers.rs

```diff
@@ -18,18 +13,15 @@ impl CounterProvider {
         let counter = Counter::new(counter_address, provider);
         Self { counter }
     }
-}
-
-#[async_trait]
-impl StateProvider for CounterProvider {
-    type State = CounterState;
 
-    async fn get_current_state(&self) -> Result<CounterState> {
-        let current_state = self.counter.number().call().await?;
-        Ok(DefaultState(current_state._0.to::<u64>()))
+    /// Reads the current on-chain number as the round.
+    pub async fn get_current_round(&self) -> Result<u64> {
+        let current = self.counter.number().call().await?;
+        Ok(current._0.to::<u64>())
     }
 
-    async fn encode_state(&self, state: &CounterState) -> Vec<u8> {
-        U256::from(state.0).abi_encode()
+    /// Encodes the round into ABI-encoded bytes for hashing/signing.
+    pub fn encode_round(&self, round: u64) -> Vec<u8> {
+        U256::from(round).abi_encode()
```

Comment:
we can move this one as well becuase it's solidity specific to the usecase folder

Reply:
this is the usecase folder